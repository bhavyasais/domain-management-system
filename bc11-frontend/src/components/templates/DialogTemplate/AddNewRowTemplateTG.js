import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import { Typography } from "@material-ui/core";

export default function AddNewRowTemplateTG(props) {

    const handleAddClick = () => {
        props.onAddClick({name,trust_score,relationship});
        props.handleClose();
    }

   
    const [name,setDomain]= useState('');
    const [trust_score,setTrustScore]= useState('');
    const [relationship,setRelationship]= useState('');

    
    const handleDomainChange = (e) => {
        setDomain(e.target.value);
    }

    const handleTrustScoreChange = (e) => {
        setTrustScore(e.target.value);
    }

    const handleRelationshipChange = (e) => {
      setRelationship(e.target.value);
    }


  return (
    <div>
      <Dialog
        open={props.open}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title"><Typography variant="h6">Add New Row</Typography></DialogTitle>
        <DialogContent>
          <form>
          
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Domain"
                variant="outlined"
                onChange={handleDomainChange}
              />
            </div>
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Trust Score"
                variant="outlined"
                onChange={handleTrustScoreChange}
              />
            </div>
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Relationship"
                variant="outlined"
                onChange={handleRelationshipChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
        <ButtonAtom
          
          variant="outlined"
          value="Close"
          textVariant="caption"
          onClick={props.handleClose}
        />
        <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        
           
           
        </DialogActions>
      </Dialog>
    </div>
  );
}