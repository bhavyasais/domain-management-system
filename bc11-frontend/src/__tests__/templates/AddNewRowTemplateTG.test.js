import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import AddNewRowTemplateTG from '../../components/templates/DialogTemplate/AddNewRowTemplateTG';
import { TextField } from '@material-ui/core';

describe('<AddNewRowTemplateTG/>', () => {
  let mounted;
  const handleClose = jest.fn();
  const onAddClick = jest.fn();

  beforeEach(() => {
    act(() => {
      mounted = mount(
        <AddNewRowTemplateTG
          handleClose={handleClose}
          onAddClick={onAddClick}
          open={true}
          onChange={onAddClick}
        />,
      );
    });
  });

  it('Renders correctly', () => {
    let wrapper;
    wrapper = renderer.create(<AddNewRowTemplateTG />);
    expect(wrapper).toMatchSnapshot();
  });
  it('handles the change', () => {
    const wrapper = mount(
      <AddNewRowTemplateTG onClick={onAddClick} handleClose={handleClose} onAddClick={onAddClick} open={true} onChange={onAddClick}/>,
    );
    wrapper.find('ButtonAtom').at(1).props().onClick();
    wrapper.find(TextField).at(0).props().onChange({target:{value:"abc"}});
    wrapper.find(TextField).at(1).props().onChange({target:{value:"abc"}});
    wrapper.find(TextField).at(2).props().onChange({target:{value:"abc"}});
  });
});
