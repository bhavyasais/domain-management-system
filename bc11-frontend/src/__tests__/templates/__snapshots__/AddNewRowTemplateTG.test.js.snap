// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddNewRowTemplateTG/> Renders correctly 1`] = `<div />`;
